src/benchmark.rs

```diff
@@ -0,0 +1,94 @@
+//! Benchmark features to implement different specific benchmarks.
+//!
+//! This is useful for creating different benchmark on time of execution or
+//! the number of instruction for example.
+
+use core::arch::asm;
+
+use crate::arch::{Arch, Architecture, Csr};
+use crate::config;
+use crate::platform::{Plat, Platform};
+
+#[macro_export]
+macro_rules! print {
```

Comment:
We don't really want to expose a normal `print`, so maybe rename it to something that makes it obvious that this macro should not be used by itself (like `_benchmark_print`)